spec.html

```diff
@@ -4409,7 +4405,29 @@ <h1>Runtime Semantics: MV</h1>
               The MV of <emu-grammar>StrUnsignedDecimalLiteral ::: DecimalDigits ExponentPart</emu-grammar> is the MV of |DecimalDigits| times 10<sup>_e_</sup>, where _e_ is the MV of |ExponentPart|.
             </li>
           </ul>
-          <p>Once the exact MV for a String numeric literal has been determined, it is then rounded to a value of the Number type. If the MV is 0, then the rounded value is *+0*<sub>𝔽</sub> unless the first non white space code point in the String numeric literal is `-`, in which case the rounded value is *-0*<sub>𝔽</sub>. Otherwise, the rounded value must be the Number value for the MV (in the sense defined in <emu-xref href="#sec-ecmascript-language-types-number-type"></emu-xref>), unless the literal includes a |StrUnsignedDecimalLiteral| and the literal has more than 20 significant digits, in which case the Number value may be either the Number value for the MV of a literal produced by replacing each significant digit after the 20th with a 0 digit or the Number value for the MV of a literal produced by replacing each significant digit after the 20th with a 0 digit and then incrementing the literal at the 20th digit position. A digit is significant if it is not part of an |ExponentPart| and</p>
+          <p>For the MV of any production not shown above, see <emu-xref href="#sec-static-semantics-mv"></emu-xref>.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-stringtonumber" aoid="StringToNumber">
+          <h1>StringToNumber ( _str_ )</h1>
+          <p>The abstract operation StringToNumber takes argument _str_ (a String) and returns a Number. It performs the following steps when called:</p>
+          <emu-alg>
+            1. Let _text_ be ! StringToCodePoints(_str_).
+            1. Let _literal_ be ParseText(_text_, |StringNumericLiteral|).
+            1. If _literal_ is a List of errors, return *NaN*.
+            1. If _literal_ contains a |StrUnsignedDecimalLiteral| and _literal_ has more than 20 significant digits, then
+              1. Let _lit_ be an implementation-defined choice of:
+                * _literal_
+                * a literal produced from _literal_ by replacing each significant digit after the 20th with a `0` digit
+                * a literal produced from _literal_ by replacing each significant digit after the 20th with a `0` digit and then incrementing the literal at the 20th significant digit position
+            1. Else,
+              1. Let _lit_ be _literal_.
+            1. Let _mv_ be the MV of _lit_.
+            1. If _mv_ is 0 and the first non white space code point in _text_ is `-`, return *-0*<sub>𝔽</sub>.
```

Comment:
Actually, consider the expression `Number("\n-0")`. Presumably, the intent is that "the first non white space code point in _text_ is `-`", and so negative zero is returned. But technically, the first "non white space" code point is the line feed (because line terminators aren't white space), so the condition in question would be false, and positive zero would be returned.

To prevent the latter reading, I'll change `non white space` to `non-|StrWhiteSpaceChar|`, since `StrWhiteSpaceChar ::: WhiteSpace | LineTerminator`.

(Properly, there should also be an "(if any)" tossed in, because a StringNumericLiteral might have no such code point.)

Reply:
Line terminators are already whitespace, in terms of the term itself. "whitespace" here doesn't refer to `WhiteSpace`, the grammar construct.

Reply:
[12.2 White Space](https://tc39.es/ecma262/#sec-white-space) defines both "white space" and _WhiteSpace_ to not include "line terminator" / _LineTerminator_. The term "whitespace" is not defined, but the simplest interpretation is that it means the same as "white space" / _WhiteSpace_. Moreover, that interpretation is consistent with the few uses in the spec.

Of course, if you want to define a term that includes both white space and line terminators, that'd be great. You could even call it "white space" / _WhiteSpace_ as long as you renamed the current white space / _WhiteSpace_ to something distinct. But all that is outside this PR.